feature/bgp/gracefulrestart/ate_tests/err/README.md

```diff
@@ -0,0 +1,289 @@
+
+**RT-1.35.3**
+
+  ***`a. Restarting DUT speaker whose BGP process was killed abruptly. In this case ERR policy is attached to the BGP neighborship.`***
+  * use `gNOI.killProcessRequest_Signal_KILL` this time as per `gNOI proto`
+  * Expected behavior is the same as RT-1.35.2.a<br>
+ 
+  ***`b. Restarting DUT speaker whose BGP process was killed abruptly after ERR policy was removed.`***
+  * Expected behavior is the same as RT-1.35.2.b<br><br><br>
+
+   
+
+  
+
+**RT-1.35.4**
+
+  ***`a. DUT Helper for a restarting EBGP speaker whose BGP process was gracefully killed. In this case ERR policy is attached to the BGP neighborship.`***
+  * Send Graceful restart trigger from ATE:Port1
+  * Expected behavior is the same as RT-1.35.2.a
+  * Repeat by sending Graceful restart trigger from ATE:Port2 with the same expected behavior as RT-1.35.2.a<br>
+
+  ***`b. DUT Helper for a restarting EBGP speaker whose BGP process was gracefully killed after ERR policy was removed.`***
+  * Send Graceful restart trigger from ATE:Port1
+  * Expected behavior is the same as RT-1.35.2.b
+  * Repeat by sending Graceful restart trigger from ATE:Port2 with the same expected behavior as RT-1.35.2.b<br><br><br>
+  
+**RT-1.35.5**
+
+  ***`a. DUT Helper for a restarting EBGP speaker whose BGP process was killed abruptly. In this case ERR policy is attached to the BGP neighborship.`***
+  * Start traffic. Send `gNOI.killProcessRequest_Signal_KILL` as per `gNOI proto` to ATE:Port1 to stop its BGP process abruptly. Configure ATE:Port1 to delay the BGP reestablishment for 330 secs over the Hold-time. Expected behavior in this case is the same as RT-1.35.2.a
+  * Post 330Secs over Hold-time expiry, BGP on ATE:Port1 is expected to be up and all traffic is expected to be successful.<br>
+  * Repeat the same test on ATE:Port2
+  
+  ***`b. DUT Helper for a restarting EBGP speaker whose BGP process was killed abruptly after ERR policy was removed.`***
+  * Start traffic. Send `gNOI.killProcessRequest_Signal_KILL` as per `gNOI proto` to ATE:Port1 to stop its BGP process abruptly. Configure ATE:Port1 to delay the BGP reestablishment for 330 secs over the Hold-time. Expected behavior in this case is the same as RT-1.35.2.b
+  * Post 330Secs over Hold-time expiry, BGP on ATE:Port1 is expected to be up and all traffic is expected to be successful.
+  * Repeat the same test on ATE:Port2<br><br><br>
+
+   
+
+ 
+**RT-1.35.6**
+  
+```TODO: gNOI.ClearBGPNeighborRequest_GRACEFUL used in this case is under review in https://github.com/openconfig/gnoi/pull/214```
+  
+  ***`a. Expected behavior when "Administrative Reset" Notification (rfc4486) sent to the peer while the ERR policy is attached to the neighborship.`***
```

Comment:
In this scenario -- who is implementing ERR? It seems like not the DUT? Are we expecting that the DUT does ERR even when it itself triggers the restart?

Reply:
I think so. Intent of sending code 6 subcode 4 while peers have exchanged the "N" bit in the RESET Flag of Capability 64, is for both peers to reset the TCP session but not flush routes until the configured Retention period. They will add the other attributes matching the ACCEPT action of the ERR policy for routes learnt from each other and advertise accordingly to their respective peers. Tl;dr: Both on either side of the disrupted TCP will need to practice ERR for seamless operation of the impacted routes as expected.